discovery/endorsement/endorsement.go

```diff
@@ -150,23 +171,51 @@ func (ea *endorsementAnalyzer) computeEndorsementResponse(ctx *context) (*discov
 
 	layouts := computeLayouts(ctx.principalsSets, principalGroups, satGraph)
 	if len(layouts) == 0 {
-		return nil, errors.New("cannot satisfy any principal combination")
+		return nil, errors.New("no peer combination can satisfy the endorsement policy")
 	}
 
 	criteria := &peerMembershipCriteria{
-		possibleLayouts: layouts,
-		satGraph:        satGraph,
-		chanMemberById:  ctx.channelMembersById,
-		idOfMembers:     ctx.identitiesOfMembers,
+		possibleLayouts:  layouts,
+		satGraph:         satGraph,
+		chanMemberById:   ctx.channelMembersById,
+		idOfMembers:      ctx.identitiesOfMembers,
+		chaincodeMapping: ctx.chaincodeMapping,
+	}
+
+	endorsersByGroup := endorsersByGroup(criteria)
```

Comment:
I personally found this confusing - variable with the same name as a function. Up to you whether you may want to rename this

Reply:
done